pkg/cmd/cockroach-sql/main.go

```diff
@@ -65,6 +65,13 @@ func main() {
 	f.BoolVar(&cfg.ConnCtx.DebugMode, cliflags.CliDebugMode.Name, cfg.ConnCtx.DebugMode, cliflags.CliDebugMode.Description)
 	f.BoolVar(&cfg.ConnCtx.Echo, cliflags.EchoSQL.Name, cfg.ConnCtx.Echo, cliflags.EchoSQL.Description)
 
+	// We want to simplify tutorials, docs etc by allowing
+	// 'cockroach-sql' to be symlinked to 'cockroach' and
+	// ensure that the resulting symlink still works when invoked
+	// as 'cockroach sql' (with a 'sql' sub-command).
+	subCmd := *sqlCmd
```

Comment:
is there a way we can add a regression test for this behaviour (a la `.tcl` test or otherwise?)

Reply:
Yeah that would be nice. But it's not easy in this PR because we need new CI rules. Will track in  #80921.